### DISCUSSION TITLE
reTerminal E10-1 | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#357] [Q&A]
# reTerminal E10-1 | Seeed Studio Wiki

reTerminal E10-1

https://wiki.seeedstudio.com/reTerminalBridge/

**Comment by jajberni** [comment]
Is there any way to add a USB3 port through M2 or PCIe? I know there are adapter cards for both interfaces, but I am not sure if they will be compatible. Certain devices (e.g. stereo cameras) require USB3 and the lack of USB3 in the reTerminal is a deal-breaker for such applications.

**Reply by MatthewJeffson** [reply]
Hi, I will inform your idea to my colleague to see their opinions.
Thanks!

**Comment by realtime3392** [comment]
What pins on the mPCIe interface are actually connected? USB? PCIe x1? etc. How do these pins map to the pins on the CM4 module?

**Reply by realtime3392** [reply]
Related question: does the mPCIe interface support mSATA storage (SSD) cards?

**Reply by realtime3392** [reply]
Page "16 mini PCI-E" of the schematic seems to indicate that the mPCIe slot only has signals routed to it that would be used by mini-PCIe communication and/or GNSS cards. For example: PCIe_1PPS, an SPI bus, connections to the SIM slot and USB pins.

Are there no PCIe x1 connections to the mini-PCIe slot? For example: are the PCIe clock, and the differential data lines **not** connected?

**Comment by realtime3392** [comment]
What pins on the M.2 interface are connected? Are only M.2 SATA Key B modules supported, or does the interface support NVMe for Key B+M cards? How do these pins mapp to the pins on the CM4 module?

**Reply by realtime3392** [reply]
OK, page "17 M.2 B Key" appears to suggest that the M.2 Key B slot is for SATA SSDs only. Is that correct?

The "CONFIG bit 0..3" section on that page also seems to indicate that SSD and WWAN_USB3.0 are separate exclusive functions of the M.2 Key B slot.

**Comment by realtime3392** [comment]
When the "switch next to the fan" is switched between PCIE and M.2, are all of the signals removed from the de-selected slot?

I assume the M.2 slot and the mPCIe slot are mutually exclusive, and there is no way to use both of them simultaneously, even if the interfaces needed by the two cards do not overlap. For example: a USB WWAN card in the mPCIe slot (that does not use the PCIe signals at all) and a Key B SATA SSD in the M.2 slot (which does not use the USB signals at all).

**Reply by realtime3392** [reply]
Page "15 I/O Expander_SIM Card" seems to indicate that the DPDT switch is only used to route the SIM slot pins to either the mPCIe slot or the M.2 slot. Are those the only pins that are affected by the "PCIE" or "M.2" switch?

**Reply by nfs0619** [reply]
Thanks for reaching out! For assistance with this issue, we recommend contacting our technical support team at **[techsupport@seeed.io](mailto:techsupport@seeed.io)**. Please provide them with the error details and your setup information, and they will assist you further.

**Reply by realtime3392** [reply]
There are no error details. I just wanted to ask the community for more details related to the expansion board. Thanks!
